### PR TITLE
web: update to ESLint 9

### PR DESCRIPTION
**Pull request by kensternberg-authentik**
ESLint 9 has been out for awhile now, and all of the plug-ins that we use have caught up, so it is time to bite the bullet and upgrade.  This commit:

- upgrades to ESLint 9, and upgrades all associated plugins
- Replaces the `.eslintrc` and `.eslintignore` files with the new, "flat" configuration file, "eslint.config.mjs".
- Places the previous "precommit" and "nightmare" rules in `./scripts/eslint.precommit.mjs` and `./scripts/eslint.nightmare.mjs`, respectively
- Replaces the scripted wrappers for eslint (`eslint`, `eslint-precommit`) with a single executable that takes the arguments `--precommit`, which applies a stricter set of rules, and `--nightmare`, which applies an even more terrifyingly strict set of rules.
- Provides the scripted wrapper `./scripts/eslint.mjs` so that eslint can be run from `bun`, if one so chooses.
- Fixes *all* of the lint `eslint.config.mjs` now finds, including removing all of the `eslint` styling rules and overrides because Eslint now proudly leaves that entirely up to Prettier.

To shut Dependabot up about ESLint.

-   [X] The code has been formatted (`make web`)

### CODE REVIEW

File: web/sfe/index.ts [hunk 1]
```diff
@@ -407,14 +405,11 @@ class AuthenticatorValidateStage extends Stage<AuthenticatorValidationChallenge>
     }
 
     renderChallengePicker() {
-        const challenges = this.challenge.deviceChallenges.filter((challenge) => {
```
Comment (kensternberg-authentik) [inline web/sfe/index.ts:410]: eslint/merge-successive-ifs

File: web/sfe/index.ts [hunk 2]
```diff
@@ -354,12 +354,10 @@ class AuthenticatorValidateStage extends Stage<AuthenticatorValidationChallenge>
             },
         );
 
-        const transformedCredentialRequestOptions = Object.assign({}, credentialRequestOptions, {
+        return Object.assign({}, credentialRequestOptions, {
```
Comment (kensternberg-authentik) [inline web/sfe/index.ts:357]: eslint/no-useless-vars

File: web/sfe/index.ts [hunk 3]
```diff
@@ -309,12 +311,10 @@ class AuthenticatorValidateStage extends Stage<AuthenticatorValidationChallenge>
         user.id = this.u8arr(this.b64enc(this.u8arr(stringId)));
         const challenge = this.u8arr(credentialCreateOptions.challenge.toString());
 
-        const transformedCredentialCreateOptions = Object.assign({}, credentialCreateOptions, {
+        return Object.assign({}, credentialCreateOptions, {
```
Comment (kensternberg-authentik) [inline web/sfe/index.ts:314]: eslint/no-useless-vars

File: web/sfe/index.ts [hunk 4]
```diff
@@ -173,7 +175,7 @@ class IdentificationStage extends Stage<IdentificationChallenge> {
                 ${
                     this.challenge.passwordFields
                         ? `<div class="form-label-group my-3 has-validation">
-                                <input type="password" class="form-control ${this.error("password").length > 0 ? "is-invalid" : ""}" name="password" placeholder="Password">
+                                <input type="password" class="form-control ${this.error("password").length > 0 ? IS_INVALID : ""}" name="password" placeholder="Password">
```
Comment (kensternberg-authentik) [inline web/sfe/index.ts:178]: eslint/no-duplicate-tokens

File: web/src/components/ak-multi-select.ts [hunk 5]
```diff
@@ -94,6 +89,11 @@ export class AkMultiSelect extends AkControlElement {
         return this.values;
     }
 
+    connectedCallback() {
```
Comment (kensternberg-authentik) [inline web/src/components/ak-multi-select.ts:92]: wc/no-attribute-access-in-constructor

File: web/src/elements/CodeMirror.ts [hunk 6]
```diff
@@ -73,9 +73,10 @@ export class CodeMirrorTextarea<T> extends AKElement {
     }
 
     @property()
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any,@typescript-eslint/explicit-module-boundary-types
     set value(v: T | string) {
-        if (v === null || v === undefined) return;
+        if (v === null || v === undefined) {
```
Comment (kensternberg-authentik) [inline web/src/elements/CodeMirror.ts:77]: eslint/no-lonely-if

File: web/src/elements/forms/Form.ts [hunk 7]
```diff
@@ -48,7 +48,9 @@ function assignValue(element: HTMLNamedElement, value: unknown, json: KeyUnknown
     for (let index = 0; index < nameElements.length - 1; index++) {
         const nameEl = nameElements[index];
         // Ensure all nested structures exist
-        if (!(nameEl in parent)) parent[nameEl] = {};
+        if (!(nameEl in parent)) {
```
Comment (kensternberg-authentik) [inline web/src/elements/forms/Form.ts:51]: eslint/no-lonely-if

File: web/src/elements/forms/Form.ts [hunk 8]
```diff
@@ -188,7 +193,7 @@ export abstract class Form<T> extends AKElement {
      */
     get isInViewport(): boolean {
         const rect = this.getBoundingClientRect();
-        return !(rect.x + rect.y + rect.width + rect.height === 0);
+        return rect.x + rect.y + rect.width + rect.height !== 0;
```
Comment (kensternberg-authentik) [inline web/src/elements/forms/Form.ts:196]: eslint/prefer-no-negated-expressions

File: web/src/elements/forms/Form.ts [hunk 9]
```diff
@@ -304,9 +308,14 @@ export abstract class Form<T> extends AKElement {
         } catch (ex) {
             if (ex instanceof ResponseError) {
                 let msg = ex.response.statusText;
-                if (ex.response.status > 399 && ex.response.status < 500) {
+                if (
+                    ex.response.status >= HTTP_BAD_REQUEST &&
+                    ex.response.status < HTTP_INTERNAL_SERVICE_ERROR
```
Comment (kensternberg-authentik) [inline web/src/elements/forms/Form.ts:313]: eslint/no-magic-numbers

File: web/src/elements/forms/Form.ts [hunk 10]
```diff
@@ -304,9 +308,14 @@ export abstract class Form<T> extends AKElement {
...
+                ) {
                     const errorMessage = ValidationErrorFromJSON(await ex.response.json());
-                    if (!errorMessage) return errorMessage;
+                    if (!errorMessage) {
```
Comment (kensternberg-authentik) [inline web/src/elements/forms/Form.ts:316]: eslint/no-lonely-if

File: web/src/elements/forms/Form.ts [hunk 11]
```diff
@@ -318,7 +327,9 @@ export abstract class Form<T> extends AKElement {
                     elements.forEach((element) => {
                         element.requestUpdate();
                         const elementName = element.name;
-                        if (!elementName) return;
+                        if (!elementName) {
```
Comment (kensternberg-authentik) [inline web/src/elements/forms/Form.ts:330]: eslint/no-lonely-if

File: web/src/elements/forms/SearchSelect/ak-search-select-view.ts [hunk 12]
```diff
@@ -264,21 +265,6 @@ export class SearchSelectView extends AKElement {
     }
 }
 
-type Pair = [string, string];
```
Comment (kensternberg-authentik) [inline web/src/elements/forms/SearchSelect/ak-search-select-view.ts:267]: eslint/no-use-before-define

Moving this into a `./utils.ts` folder helped, and helped keep this cleaner.

File: web/src/elements/forms/SearchSelect/ak-search-select.ts [hunk 13]
```diff
@@ -97,11 +97,6 @@ export class SearchSelect<T> extends CustomEmitterElement(AkControlElement) {
     @state()
     error?: APIErrorTypes;
 
-    constructor() {
-        super();
-        this.dataset.akControl = "true";
```
Comment (kensternberg-authentik) [inline web/src/elements/forms/SearchSelect/ak-search-select.ts:102]: wc/no-attribute-accessor-in-constructor
eslint/no-useless-constructor

File: web/src/user/LibraryPage/ak-library-impl.ts [hunk 14]
```diff
@@ -104,7 +104,7 @@ export class LibraryPage extends AKElement {
     searchUpdated(event: LibraryPageSearchUpdated) {
         event.stopPropagation();
         const apps = event.apps;
-        if (!(apps.length > 0)) {
+        if (apps.length <= 0) {
```
Comment (kensternberg-authentik) [inline web/src/user/LibraryPage/ak-library-impl.ts:107]: eslint/prefer-no-negated-expressions

File: web/src/user/UserInterface.ts [hunk 15]
```diff
@@ -169,7 +168,7 @@ class UserInterfacePresentation extends AKElement {
     }
 
     get isFullyConfigured() {
-        return !!(this.uiConfig && this.me && this.brand);
+        return Boolean(this.uiConfig && this.me && this.brand);
```
Comment (kensternberg-authentik) [inline web/src/user/UserInterface.ts:171]: eslint/no-extra-boolean-cast

File: web/src/user/UserInterface.ts [hunk 16]
```diff
@@ -455,7 +454,7 @@ export class UserInterface extends EnterpriseAwareInterface {
     }
 
     get isFullyConfigured() {
-        return !!(this.uiConfig && this.me);
+        return Boolean(this.uiConfig && this.me);
```
Comment (kensternberg-authentik) [inline web/src/user/UserInterface.ts:457]: eslint/no-extra-boolean-cast

**Review by kensternberg-authentik — COMMENTED** [review]
Just left a few comments on why eslint encouraged a change here and there.

**Review by BeryJu — APPROVED** [review]

**Pull request merged**